synapse/lib/ast.py

```diff
@@ -3060,6 +3066,10 @@ async def getPropAndValu(self, runt, path):
                                                     name=name, form=path.node.form.name))
 
             valu = path.node.get(name)
+            if isinstance(valu, dict):
```

Comment:
This could also be a list which could be mutated.  May need to check for dict, list, or tuple.

Reply:
a tuple can contain a nested list or dict so you'll still need to do this to prevent the behavior.